packages/frontend/components/Plan/DuplicatePlanButton.tsx

```diff
@@ -0,0 +1,111 @@
+import { CopyIcon } from "@chakra-ui/icons";
+import { API } from "@graduate/api-client";
+import { CreatePlanDto, PlanModel } from "@graduate/common";
+import { useRouter } from "next/router";
+import { Dispatch, SetStateAction, useState, useContext } from "react";
+import { toast } from "react-toastify";
+import { mutate } from "swr";
+import {
+  USE_STUDENT_WITH_PLANS_SWR_KEY,
+  useStudentWithPlans,
+} from "../../hooks";
+import {
+  cleanDndIdsFromPlan,
+  cleanDndIdsFromStudent,
+  handleApiClientError,
+} from "../../utils";
+import { BlueButton } from "../Button";
+import { IsGuestContext } from "../../pages/_app";
+import { getLocalPlansLength } from "../../utils/plan/getLocalPlansLength";
+import { GraduateToolTip } from "../GraduateTooltip";
+
+interface DuplicatePlanButton {
+  plan: PlanModel<string>;
+  setSelectedPlanId: Dispatch<SetStateAction<number | undefined | null>>;
+}
+
+export const DuplicatePlanButton: React.FC<DuplicatePlanButton> = ({
+  plan,
+  setSelectedPlanId,
+}) => {
+  const router = useRouter();
+  const { isGuest } = useContext(IsGuestContext);
+  const [buttonLoading, setButtonLoading] = useState(false);
+  const { student } = useStudentWithPlans();
+
+  const duplicatePlan = async () => {
+    if (!student) return;
+    // TODO: figure out when to do with this -- debounce stuff
```

Comment:
is this TODO still relevant? i think adding the plan limit was our solution, right?